akka-http/src/main/scala/akka/http/scaladsl/server/directives/ParameterDirectives.scala

```diff
@@ -125,7 +125,10 @@ object ParameterDirectives extends ParameterDirectives {
       extractRequestContext flatMap { ctx ⇒
         import ctx.executionContext
         import ctx.materializer
-        handleParamResult(paramName, fsou(ctx.request.uri.query().get(paramName)))
+        Try(ctx.request.uri.query()) match {
+          case Success(query) ⇒ handleParamResult(paramName, fsou(query.get(paramName)))
+          case Failure(t)     ⇒ reject(MalformedRequestContentRejection("The request's query string is invalid", t))
```

Comment:
Should we add the message to the string as well, at least if we know that it is a parser error, which we surface to the client at other places?